src/http.rs

```diff
@@ -382,8 +386,165 @@ pub fn build_http_trace_layer(
         .on_eos(())
 }
 
+/// Configuration of HTTP server keepalive parameters.
+#[serde_as]
+#[configurable_component]
+#[derive(Clone, Debug, PartialEq)]
+#[serde(deny_unknown_fields)]
+pub struct KeepaliveConfig {
+    /// The maximum amount of time a connection may exist before it is closed
+    /// by sending a `Connection: close` header on the HTTP response.
+    ///
+    /// A random jitter configured by `max_connection_age_jitter_factor` is added
+    /// to the specified duration to spread out connection storms.
+    ///
+    /// A value of `0` disables this feature.
+    #[serde_as(as = "serde_with::DurationSeconds<u64>")]
+    #[serde(
+        rename = "max_connection_age_secs",
+        default = "default_max_connection_age"
+    )]
+    #[configurable(metadata(docs::examples = 600))]
+    #[configurable(metadata(docs::human_name = "Maximum Connection Age"))]
+    pub max_connection_age: Duration,
+
+    /// The factor by which to jitter the `max_connection_age_secs` value.
+    ///
+    /// A value of 0.1 means that the actual duration will be between 90% and 110% of the
+    /// specified duration.
+    #[serde(default = "default_max_connection_age_jitter_factor")]
+    #[configurable(validation(range(min = 0.0, max = 1.0)))]
+    pub max_connection_age_jitter_factor: f64,
+}
+
+const fn default_max_connection_age() -> Duration {
+    Duration::from_secs(0)
+}
```

Comment:
Do you think it'd be useful for users to configure Vector to close after every connection? Just wondering if `0` could be used for that and we instead have another option to disable the feature (similar to the `--graceful-shutdown-limit-secs` and `--no-graceful-shutdown-limit` options).

Reply:
Hmm good question. I chose this design over using an option because of the nice serde helper to automatically convert the input to a `Duration`.  We can use an option if we think that behavior is preferable from a UX POV.  I imagine someone could come up with a use-case for close after every connection but that seems very inefficient.. From a UX POV, I think that functionality would fit better under a `max_connection_requests` param option.

Reply:
Ah, yeah, that's fair. Maybe that would fit better as `max_connection_requests`. I'll note that we did codify a rule to avoid "sentinel values" for flags:

https://github.com/vectordotdev/vector/blob/master/docs/specs/configuration.md#flag-naming

We _could_ extend that rule to configuration options too. I have a concern that the UX will be inconsistent for users, without a rule, and we'll have to answer this question each time we add a new option. We could codify the same rule we have for flags and/or define a new one for configuration options.

Reply:
I'll update the value from `Duration` to `Option<Duration` to avoid the sentinel value, which I agree is a valid point of confusion from a UX perspective.

Reply:
Good thought 👍 Granted TOML still doesn't have an answer for "null" values, but YAML and JSON users should be able to use `null` to indicate no limit.

Reply:
Done!